sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -303,6 +295,66 @@ case class Hex(child: Expression) extends UnaryExpression with Serializable  {
   }
 }
 
+/**
+ * Performs the inverse operation of HEX.
+ * Resulting characters are returned as a byte array.
+ */
+case class Unhex(child: Expression)
+  extends UnaryExpression with AutoCastInputTypes with Serializable {
+
+  override def nullable: Boolean = true
+  override def dataType: DataType = BinaryType
+  override def inputTypes: Seq[DataType] = Seq(BinaryType)
+
+  override def eval(input: InternalRow): Any = {
+    val num = child.eval(input)
+    if (num == null) {
+      null
+    } else {
+      unhex(num.asInstanceOf[UTF8String].getBytes)
+    }
+  }
+
+  // lookup table to translate '0' -> 0 ... 'F'/'f' -> 15
+  private[this] val unhexDigits = {
```

Comment:
why would it be more clear? are you thinking about the distance between its definition and where it is used?

this is one case where java beats scala with static fields. Ideally all the string functions should just be member functions in UTF8String.